examples/examples.md

```diff
@@ -0,0 +1,297 @@
+                           nullptr,     // data
+                           nullptr,     // signature
+                           "1",         // chain id
+                            1,          // version
+                           nullptr);    // options
+                           
+    // Get serialized transaction
+    std::string const txSerialized = transaction.serialize(); 
+    
+    // Deserialize transaction into current object
+    transaction.deserialize("{\"nonce\":0,\"value\":\"0\",\"receiver\":\"erd1...}");
+    
+    // Sign/verify transaction
+    Signer signer(seed); 
+    transaction.sign(signer);
+    if (!transaction.verify()){
+        // Something went wrong
+    }
+}
+catch (std::invalid_argument const &err)
+{
+    // Invalid argument, e.g.:
+    // -Missing/invalid transaction field
+    // -Invalid serialized transaction
+}
+catch (std::length_error const &err)
+{
+    // ...  
+}
+
+```
+
+### 1.1.5 Proxy
+
+```c++
+try
+{
+    ProxyProvider proxy("https://testnet-gateway.elrond.com");
+    Address const address("erd1qq...");
+    Account const account = proxy.getAccount(address);
+}
+catch (std::invalid_argument const &err)
+{
+    // ...
+}
+catch (std::runtime_error const &err)
+{
+    // ...  
+}
+
+```
+
+### 1.1.6 ESDT transfer with func call
+
+```c++
+try
+{
+    // Get address from json
+    KeyFileReader keyFile("alice.json", "password");
+    Address const address = keyFile.getAddress();
+    
+    // Load account from proxy
+    ProxyProvider proxy("https://testnet-gateway.elrond.com");
+    Account const account = proxy.getAccount(address);
+    
+    // Create transaction
+    Transaction transaction;
+    transaction.m_sender = std::make_shared<Address>(account);
+    transaction.m_receiver = std::make_shared<Address>("erd1qqq..."); // contract address
+    transaction.m_chainID = "T";
+    transaction.m_nonce = account.getNonce();
+    transaction.m_gasPrice = 1000000000;
+    transaction.m_gasLimit = 150000000;
+
+    // Define function args
+    SCArguments args;
+    args.add("param1");
+    args.add(BigUInt("400000"));
+    
+    // Prepare transaction for ESDT transfer with func call + arguments
+    prepareTransactionForESDTTransfer(transaction, 
+                                      "TOKEN-b9cba1", // token id to transfer
+                                      "func",         // function to call
+                                      args);          // SC args
+                                      
+    // Sign transaction                                 
+    Signer const signer(keyFile.getSeed());
+    transaction.sign(signer);
+    
+    // Send transaction and get tx hash
+    std::string const txHash = proxy.send(transaction).hash;
+    
+    // Check tx status
+    TransactionStatus const txStatus = proxy.getTransactionStatus(txHash);
+    if (txStatus.isSuccessful())
+    {
+        // ...
+    }
+    
+}
+catch (...)
+{
+    // ...
+}
+```
+
+## 1.2 Examples. CLI
```

Comment:
`## 2.2. Examples. CLI`

Reply:
Agree, would be better. Refactored as suggested.